editor/src/clj/editor/properties_view.clj

```diff
@@ -129,46 +147,59 @@
     :script-property-type-url "script-property-text-field-icon-url"
     nil))
 
-(defmulti create-property-control! (fn [edit-type _ property-fn]
+(defmulti create-property-control! (fn [edit-type _ _]
                                      (edit-type->type edit-type)))
 
 (defmethod create-property-control! g/Str [edit-type _ property-fn]
-  (let [text         (TextField.)
+  (let [text (TextField.)
         update-ui-fn (partial update-text-fn text str)
-        update-fn    (fn [_]
-                       (properties/set-values! (property-fn) (repeat (.getText text))))]
-    (customize! text update-fn)
+        cancel-fn (fn [_]
+                    (let [property (property-fn)]
+                      (update-ui-fn (properties/values property)
+                                    (properties/validation-message property)
+                                    (properties/read-only? property))))
+        update-fn (fn [_]
+                    (properties/set-values! (property-fn) (repeat (.getText text))))]
+    (customize! text update-fn cancel-fn)
     (when-let [style-class (script-property-type->style-class (:script-property-type edit-type))]
       (add-style-class! text style-class))
     [text update-ui-fn]))
 
 (defmethod create-property-control! g/Int [edit-type _ property-fn]
-  (let [text         (TextField.)
+  (let [text (TextField.)
         update-ui-fn (partial update-text-fn text field-expression/format-int)
-        update-fn    (fn [_]
-                       (if-let [v (field-expression/to-int (.getText text))]
-                         (let [property (property-fn)]
-                           (properties/set-values! property (repeat v))
-                           (update-ui-fn (properties/values property)
-                                         (properties/validation-message property)
-                                         (properties/read-only? property)))))]
-    (customize! text update-fn)
+        update-prop-fn (fn [_]
+                      (let [property (property-fn)]
+                        (update-ui-fn (properties/values property)
+                                      (properties/validation-message property)
+                                      (properties/read-only? property))))
+        cancel-fn update-prop-fn
+        update-fn (fn [_]
+                    (if-let [v (field-expression/to-int (.getText text))]
+                      (let [property (property-fn)]
+                        (properties/set-values! property (repeat v))
+                        (update-prop-fn nil))))]
```

Comment:
in this case we call it anyway, not sure if it's by design, but I didn't change it. But I named function `update-prop-fn` instead of cancel to make it clear